pkg/webhook/webhook_test.go

```diff
@@ -127,6 +128,235 @@ var _ = Describe("Webhook", func() {
+			SetCustomizedInjections(in)
+			Expect(cusInjects.Patchs).Should(Equal(out))
+		},
+		Entry(
+			"patch - empty config map",
+			&corev1.ConfigMap{
+				Data: map[string]string{},
+			},
+			map[string]jsonPatchOperation{},
+			map[string]jsonPatchOperation{},
+		),
+		Entry(
+			"patch - addtional networks annotation",
+			&corev1.ConfigMap{
+				Data: map[string]string{
+					"nri-inject-annotation": "{\"op\": \"add\", \"path\": \"/metadata/annotations\", \"value\": {\"k8s.v1.cni.cncf.io/networks\": \"sriov-net\"}}"},
+			},
+			map[string]jsonPatchOperation{},
+			map[string]jsonPatchOperation{
+				"nri-inject-annotation": jsonPatchOperation{
+					Operation: "add",
+					Path: "/metadata/annotations",
+					Value: map[string]interface{}{"k8s.v1.cni.cncf.io/networks": "sriov-net"},
+				},
+			},
+		),
+		Entry(
+			"patch - default network annotation",
+			&corev1.ConfigMap{
+				Data: map[string]string{
+					"nri-inject-annotation": "{\"op\": \"add\", \"path\": \"/metadata/annotations\", \"value\": {\"v1.multus-cni.io/default-network\": \"sriov-net\"}}"},
+			},
+			map[string]jsonPatchOperation{},
+			map[string]jsonPatchOperation{
+				"nri-inject-annotation": jsonPatchOperation{
+					Operation: "add",
+					Path: "/metadata/annotations",
+					Value: map[string]interface{}{"v1.multus-cni.io/default-network": "sriov-net"},
+				},
+			},
+		),
+		Entry(
+			"patch - non-annotation",
+			&corev1.ConfigMap{
+				Data: map[string]string{
+					"nri-inject-labels": "{\"op\": \"add\", \"path\": \"/metadata/labels\", \"value\": {\"v1.multus-cni.io/default-network\": \"sriov-net\"}}",
+				},
+			},
+			map[string]jsonPatchOperation{},
+			map[string]jsonPatchOperation{},
+		),
+		Entry(
+			"patch - remove stale entry",
+			&corev1.ConfigMap{
+				Data: map[string]string{},
+			},
+			map[string]jsonPatchOperation{
+				"nri-inject-annotation": jsonPatchOperation{
+					Operation: "add",
+					Path: "/metadata/annotations",
+					Value: map[string]interface{}{"v1.multus-cni.io/default-network": "sriov-net"},
+				},
+			},
+			map[string]jsonPatchOperation{},
+		),
+		Entry(
+			"patch - overwrite existing cusInjects",
+			&corev1.ConfigMap{
+				Data: map[string]string{
+					"nri-inject-annotation": "{\"op\": \"add\", \"path\": \"/metadata/annotations\", \"value\": {\"v1.multus-cni.io/default-network\": \"sriov-net-new\"}}"},
+			},
+			map[string]jsonPatchOperation{
+				"nri-inject-annotation": jsonPatchOperation{
+					Operation: "add",
+					Path: "/metadata/annotations",
+					Value: map[string]interface{}{"v1.multus-cni.io/default-network": "sriov-net-old"},
+				},
+			},
+			map[string]jsonPatchOperation{
+				"nri-inject-annotation": jsonPatchOperation{
+					Operation: "add",
+					Path: "/metadata/annotations",
+					Value: map[string]interface{}{"v1.multus-cni.io/default-network": "sriov-net-new"},
+				},
+			},
+		),
+	)
+
```

Comment:
Could you add a test that ensures a patch operation fails if the Path is not prefixed with "/metadata/" ?
we should ensure this does not change unintentionally in the future.

Reply:
This unit test would fail if other path is used:
https://github.com/k8snetworkplumbingwg/network-resources-injector/pull/59/files#diff-54553ccca2e329f7bf1fdf04d002ced7323d3fa477e7fc6e2a0a9d248cb10ab2R313-R322
do you think it is enough?

Reply:
Ah, missed this one !